tcl/org.testeditor.tcl.dsl.ide/src/org/testeditor/tcl/dsl/ide/TclContentProposalProvider.xtend

```diff
@@ -0,0 +1,296 @@
+	 */
+	override protected def Predicate<IEObjectDescription> getCrossrefFilter(CrossReference reference, ContentAssistContext context) {
+		val type = currentTypeFinder.findCurrentTypeAfter(reference)
+		if (type.instanceClass == VariableReferencePathAccess) {
+			return [ eObjectDescription |
+				val eObject = eObjectDescription.EObjectOrProxy
+				return !(eObject instanceof TemplateVariable)
+			]
+		} else {
+			return super.getCrossrefFilter(reference, context)
+		}
+	}
+
+	private def makeTestStepProposals(ContentAssistContext context, IIdeContentProposalAcceptor acceptor) {
+		val entry = proposalCreator.createProposal('- ', context)
+		val prio = proposalPriorities.getDefaultPriority(entry)
+		acceptor.accept(entry, prio)
+		val model = context.currentModel
+		if (context.isLocatedWithinComponentUsage && context.isOutsidePreviousTestStep) {
+			(model as ComponentTestStepContext).component?.allInteractionTypes.forEach [
+				val proposal = '''- «template.restoreString»''' // NOTE THAT - IS EXPLICITLY INCLUDED HERE
+				val templateEntry = proposalCreator.createProposal(proposal, context)
+				if (templateEntry !== null) {
+					acceptor.accept(templateEntry, proposalPriorities.getCrossRefPriority(null, entry))
+				}
+			]
+
+		} else if (model instanceof MacroTestStepContext) {
+			model.macroCollection.makeMacroTemplateProposal(context, acceptor)
+		}
+	}
+
+	private def boolean isLocatedWithinComponentUsage(ContentAssistContext context) {
+		return (context.currentModel instanceof ComponentTestStepContext && !(context.previousModel instanceof TestStep))
+	}
+
+	private def boolean isOutsidePreviousTestStep(ContentAssistContext context) {
+		return !(context.previousModel instanceof StepContent && context.previousModel.isIncompleteTestStep)
+	}
+
+	private def void makeStepContentElementProposals(ContentAssistContext context, IIdeContentProposalAcceptor acceptor) {
+		val model = context.currentModel
+		if (model instanceof TestStep) {
+			model.envParams.forEach[makeEnvParamProposal(context, acceptor)]
+			model.enclosingMacroParameters.forEach[makeMacroParameterProposal(context, acceptor)]
+			if (model.componentContext !== null) { // since the macro context does not allow for element parameter, yet, only component contexts are relevant here
+				model.makeElementParamsProposal('', context, acceptor)
+			}
+			if (context.previousModel.isIncompleteTestStep) {
+				model.componentContext?.component?.makeComponentTemplateProposal(model, context, acceptor)
+				model.macroContext?.macroCollection?.makeMacroTemplateProposal(context, acceptor)
+			}
+			return
+		} else if (model instanceof StepContentElement) {
+			model.makeElementParamsProposal(context, acceptor)
+			return
+		}
+	}
+
+	/**
+	 * actually the last parsed element is tested whether it is part of a complete test step, thus allowing for now further step contents, 
+	 * whereas the parser does not know that and might propose additional step contents
+	 */
```

Comment:
did not understand this comment, in particular the part "_thus allowing for now further step contents_".